src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -7995,6 +7995,11 @@ private void ParseDeclarationModifiers(SyntaxListBuilder list)
                 {
                     mod = this.AddError(mod, ErrorCode.ERR_BadMemberFlag, mod.Text);
                 }
+                else if (list.Any(mod.Kind))
+                {
+                    // check for duplicates, can only be const
+                    mod = this.AddError(mod, ErrorCode.ERR_TypeExpected, mod.Text);
```

Comment:
I think it would be better to report more specific error, i.e. duplicate modifier.